### PR TITLE
feat: add custom error messaging

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Adds a markdown prop to allow links + other custom error messaging to GridForm. This also fixes a bug where GridForm error messages are taking up space.

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/rFbCZPynW35wz1afRQtLZZ/Single-Sign-On?node-id=1251%3A11120
- [x] Related to JIRA ticket: GM-292
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@42.1.2-alpha.842d6c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6100714acbdf302f37262ac8--gamut-preview.netlify.app>https://6100714acbdf302f37262ac8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6100714acbdf302f37262ac8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.3-alpha.8cc06c.0<br/>@codecademy/gamut-labs@19.16.15-alpha.8cc06c.0<br/>@codecademy/gamut@35.4.3-alpha.8cc06c.0<br/>@codecademy/styleguide@42.3.3-alpha.8cc06c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6101ab83f455f36f9bf2750c--gamut-preview.netlify.app>https://6101ab83f455f36f9bf2750c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6101ab83f455f36f9bf2750c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.3-alpha.5f0e0b.0<br/>@codecademy/gamut-labs@19.16.15-alpha.5f0e0b.0<br/>@codecademy/gamut@35.4.3-alpha.5f0e0b.0<br/>@codecademy/styleguide@42.3.3-alpha.5f0e0b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6101b31bd3a52908f1474b59--gamut-preview.netlify.app>https://6101b31bd3a52908f1474b59--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6101b31bd3a52908f1474b59>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.3-alpha.a7d54c.0<br/>@codecademy/gamut-labs@19.16.15-alpha.a7d54c.0<br/>@codecademy/gamut@35.4.3-alpha.a7d54c.0<br/>@codecademy/styleguide@42.3.3-alpha.a7d54c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6101b55bd38ed408cd2e92a4--gamut-preview.netlify.app>https://6101b55bd38ed408cd2e92a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6101b55bd38ed408cd2e92a4>Deploy Logs</a></p>

File: packages/gamut/src/GridForm/GridFormInputGroup/index.tsx [hunk 1]
```diff
@@ -140,6 +141,13 @@ export const GridFormInputGroup: React.FC<GridFormInputGroupProps> = ({
     );
   }
 
+  const errorContents = (error: string) => {
+    if (field.validation?.markdown) {
+      return <Markdown inline text={error} spacing="none" />;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/GridForm/GridFormInputGroup/index.tsx:146]: What if we... always allowed markdown?

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 2]
```diff
@@ -1023,6 +1024,37 @@ We call it a "sweet container" so that bots do not immediately detect it as a ho
   </Story>
 </Canvas>
 
+## Markdown Errors
+
+GridForm can optionally render errors through our <LinkTo id="atoms-markdown--basics">Markdown</LinkTo> component and our markdown prop. This should be used sparingly.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:1029]: > This should be used sparingly.

Why?

File: packages/gamut/src/GridForm/GridFormInputGroup/index.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/GridForm/GridFormInputGroup/index.tsx:146]: 😈

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx (hunk 2 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:1029]: ahah yeah, this is vague. there some design things about using links in errors that isn't exactly encourage but i'll pick stephen's brain about how to word this better

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.3-alpha.bcb6ce.0<br/>@codecademy/gamut-labs@19.16.15-alpha.bcb6ce.0<br/>@codecademy/gamut@35.4.3-alpha.bcb6ce.0<br/>@codecademy/styleguide@42.3.3-alpha.bcb6ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6102b50e61c2711f7c501497--gamut-preview.netlify.app>https://6102b50e61c2711f7c501497--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6102b50e61c2711f7c501497>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.4-alpha.2f29ff.0<br/>@codecademy/gamut-labs@19.16.16-alpha.2f29ff.0<br/>@codecademy/gamut@35.4.4-alpha.2f29ff.0<br/>@codecademy/styleguide@42.3.5-alpha.2f29ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6103f1d7396916a12ccd9ad8--gamut-preview.netlify.app>https://6103f1d7396916a12ccd9ad8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6103f1d7396916a12ccd9ad8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.4-alpha.cf023e.0<br/>@codecademy/gamut-labs@19.16.16-alpha.cf023e.0<br/>@codecademy/gamut@35.4.4-alpha.cf023e.0<br/>@codecademy/styleguide@42.3.5-alpha.cf023e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6103f423e8288d5f61919525--gamut-preview.netlify.app>https://6103f423e8288d5f61919525--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6103f423e8288d5f61919525>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.4-alpha.e8c8eb.0<br/>@codecademy/gamut-labs@19.16.16-alpha.e8c8eb.0<br/>@codecademy/gamut@35.4.4-alpha.e8c8eb.0<br/>@codecademy/styleguide@42.3.5-alpha.e8c8eb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://610400444b4cf75c41573dc2--gamut-preview.netlify.app>https://610400444b4cf75c41573dc2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/610400444b4cf75c41573dc2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.4-alpha.c61c35.0<br/>@codecademy/gamut-labs@19.16.16-alpha.c61c35.0<br/>@codecademy/gamut@35.4.4-alpha.c61c35.0<br/>@codecademy/styleguide@42.3.5-alpha.c61c35.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61042ecbbcf39a00902d5cc6--gamut-preview.netlify.app>https://61042ecbbcf39a00902d5cc6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61042ecbbcf39a00902d5cc6>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
🆒❗

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.4-alpha.eab0e7.0<br/>@codecademy/gamut-labs@19.16.16-alpha.eab0e7.0<br/>@codecademy/gamut@35.4.4-alpha.eab0e7.0<br/>@codecademy/styleguide@42.3.5-alpha.eab0e7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6108434c5cf4355e35e862ff--gamut-preview.netlify.app>https://6108434c5cf4355e35e862ff--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6108434c5cf4355e35e862ff>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.6-alpha.d8b104.0<br/>@codecademy/gamut-labs@19.16.18-alpha.d8b104.0<br/>@codecademy/gamut@35.5.1-alpha.d8b104.0<br/>@codecademy/styleguide@42.3.9-alpha.d8b104.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61084ce7fa1df862438a7cc4--gamut-preview.netlify.app>https://61084ce7fa1df862438a7cc4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61084ce7fa1df862438a7cc4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.6-alpha.ff760e.0<br/>@codecademy/gamut-labs@19.16.18-alpha.ff760e.0<br/>@codecademy/gamut@35.5.1-alpha.ff760e.0<br/>@codecademy/styleguide@42.3.9-alpha.ff760e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61084eeb16b0983cff6f7ac4--gamut-preview.netlify.app>https://61084eeb16b0983cff6f7ac4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61084eeb16b0983cff6f7ac4>Deploy Logs</a></p>

**Pull request merged**